VersionsParser.py

```diff
@@ -0,0 +1,34 @@
+import re
+from typing import List, Dict, Pattern, Set
+
+
+class VersionsParser:
+    def __init__(self, input_lines: List[str]):
```

Comment:
I suggest adding some simple documentation and comments for this class and its member functions.